web/src/utils/filter-definitions.ts

```diff
@@ -98,46 +98,18 @@ const k8sSingleResourceEncode = (kind: Field, namespace: Field, name: Field, res
   return `${kind}="${res.kind}"&${namespace}="${res.namespace}"&${name}="${res.name}"`;
 };
 
-const peers = (
-  base: Omit<FilterDefinition, 'encoders'>,
-  srcEncoder: FiltersEncoder,
-  dstEncoder: FiltersEncoder
-): FilterDefinition[] => {
+const withDest = (src: FilterDefinition, dstEncoder: FiltersEncoder): FilterDefinition[] => {
```

Comment:
Following [TargetedFilterId approach](https://github.com/netobserv/network-observability-console-plugin/pull/343/files#r1255697135) you can keep the previous `peers` approach that takes TargetedFilterId + srcEncoder + dstEncoder as arguments

Reply:
I feel like it's mor ecomplicated... e.g. here's what I would have:

```ts
const peers = (
  base: Omit<Omit<FilterDefinition, 'id'>, 'encoder'> & { id: TargetedFilterId },
  srcEncoder: FiltersEncoder,
  dstEncoder: FiltersEncoder
): FilterDefinition[] => {
  return [
    {
      ...base,
      id: ('src_' + base.id) as FilterId,
      category: FilterCategory.Source,
      encoder: srcEncoder
    },
    {
      ...base,
      id: ('dst_' + base.id) as FilterId,
      category: FilterCategory.Destination,
      encoder: dstEncoder
    }
  ];
};
```

The double `Omit` is a bit ugly, I felt like the new approach that takes a fully built `FilterDefinition` and mirroring it was nicer

Reply:
What about split and extend the `FilterDefinition` interface like:

```
export interface FilterPeerDefinition {
  targetedId: TargetedFilterId;
  name: string;
  component: FilterComponent;
  category: FilterCategory;
  getOptions: (value: string) => Promise<FilterOption[]>;
  validate: (value: string) => { val?: string; err?: string };
  checkCompletion?: (value: string, selected: string) => { completed: boolean; option: FilterOption };
  autoCompleteAddsQuotes?: boolean;
  hint?: string;
  examples?: string;
  placeholder?: string;
  // overlap tells if the type of entity referred to with this filter may have overlapping (duplicated)
  // flows when querying for returned traffic (back and forth) - they result in slightly more complicated queries.
  overlap: boolean;
}

export interface FilterDefinition extends FilterPeerDefinition {
  id: FilterId;
  encoder: FiltersEncoder;
}
```

Then you don't need the `Omit` part anymore. You can also use back quotes for ids to avoid the casts:
```
const peers = (
  base: FilterPeerDefinition,
  srcEncoder: FiltersEncoder,
  dstEncoder: FiltersEncoder
): FilterDefinition[] => {
  return [
    {
      ...base,
      id: `src_${base.targetedId}`,
      category: FilterCategory.Source,
      encoder: srcEncoder
    },
    {
      ...base,
      id: `dst_${base.targetedId}`,
      category: FilterCategory.Destination,
      encoder: dstEncoder 
    },
  ];
};
```

I'll let you pick what you prefer :wink: no blocks either way for me